all/environments/gym.py

```diff
@@ -6,9 +6,13 @@
 
 class GymEnvironment(Environment):
     def __init__(self, env, device=torch.device('cpu')):
-        self._name = env
+        
         if isinstance(env, str):
+            self._name = env
             env = gym.make(env)
+        else:
+            self._name = env.__class__.__name__
+
```

Comment:
Nice catch!

Reply:
Yeah, I wanted to do this in a separate PR originally, but apparently I am missing something crucial as to how PRs work here on GitHub. :D :D I will need to look into that...